File: works/11403.py

def dfs(i, v, mark, canVisit):
    for x in v[i]:
        if mark[x]: continue
        mark[x] = True
        canVisit.append(x)
        dfs(x, v, mark, canVisit)


n = int(input())
ans = [[0 for _ in range(110)] for _ in range(110)]

v = [[0 for _ in range(0)] for _ in range(n)]
for i in range(n):
    _s = input().split()
    for j in range(n):
        if _s[j] == '1': v[i].append(j)


for i in range(n):
    mark = [False for _ in range(n)]
    canVisit = []
    dfs(i, v, mark, canVisit)
    for j in canVisit: ans[i][j] = 1

for i in range(n):
    for j in range(n):
        print(ans[i][j], end=' ')
    print()

File: works/11403.kt
import java.io.BufferedReader
import java.io.InputStreamReader

var n = 0
var ans = Array(110) {Array(110) {0} }
var list = Array(110){ ArrayList<Int>() }

fun dfs(i:Int, mark:ArrayList<Boolean>, canVisit:ArrayList<Int>)
{
    for(x in list[i])
    {
        if(mark[x]) continue
        mark[x] = true
        canVisit.add(x)
        dfs(x, mark, canVisit)
    }
}

fun main()
{
    val br = BufferedReader(InputStreamReader(System.`in`))
    n = br.readLine().toInt()
    for(i in 0 until n)
    {
        var _s = br.readLine().split(' ')
        for(j in 0 until n)
        {
            if(_s[j] == "1") list[i].add(j)
        }
    }

    for(i in 0 until n)
    {
        var mark = ArrayList<Boolean>()
        for(j in 0 until n) mark.add(false)
        var canVisit = ArrayList<Int>()
        dfs(i, mark, canVisit)
        for(j in canVisit) ans[i][j] = 1
    }

    for(i in 0 until n)
    {
        for(j in 0 until n)
        {
            print(ans[i][j])
            print(" ")
        }
        println()
    }

}